File: maven-confluence-reporting-plugin/src/test/java/org/bsc/maven/reporting/test/SiteTest.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.bsc.maven.reporting.test;

import java.nio.file.Path;
import java.nio.file.Paths;

import javax.xml.bind.JAXBContext;
import javax.xml.bind.Unmarshaller;

import org.bsc.confluence.model.Site;
import org.hamcrest.core.Is;
import org.hamcrest.core.IsEqual;
import org.hamcrest.core.IsInstanceOf;
import org.hamcrest.core.IsNull;
import org.junit.Assert;
import org.junit.Ignore;
import org.junit.Test;

/**
 *
 * @author bsorrentino
 */
public class SiteTest {
    
    
    /**
     * Must be ignored because i got
     * 
     * java.lang.NoClassDefFoundError: org/bsc/maven/reporting/model/Site
     * at org.bsc.maven.reporting.test.SiteTest.jaxbTest(SiteTest.java:30)
     * 
     * @throws Exception 
     */
    @Test @Ignore
    public void jaxbTest() throws Exception {

        Path basepath = Paths.get("src/site/confluence/home.confluence");
        java.net.URI relativeURI = new java.net.URI("template.confluence");
        System.out.printf("uri=[%s]\n", basepath.toUri().resolve(relativeURI) );
        
        JAXBContext jc = JAXBContext.newInstance(Site.class);
        
        Unmarshaller unmarshaller = jc.createUnmarshaller();

        Object result = unmarshaller.unmarshal( getClass().getClassLoader().getResourceAsStream("site.xml"));

        Assert.assertThat(result, IsNull.notNullValue());
        Assert.assertThat(result, IsInstanceOf.instanceOf(Site.class));
        
        Site site = (Site) result;
        
        site.setBasedir( basepath );
        Assert.assertThat(site.getHome().getUri(), IsNull.notNullValue());        
        Assert.assertThat(site.getHome().getName(), IsEqual.equalTo("home"));        
        //Assert.assertThat(site.getHome().getSource().getName(), IsEqual.equalTo("home.confluence"));        
        Assert.assertThat(site.getHome(), IsNull.notNullValue());        
        Assert.assertThat(site.getHome().getAttachments(), IsNull.notNullValue());        
        Assert.assertThat(site.getHome().getAttachments().isEmpty(), Is.is(false));
        Assert.assertThat(site.getHome().getChildren(), IsNull.notNullValue());        
        Assert.assertThat(site.getHome().getChildren().isEmpty(), Is.is(false));
        Assert.assertThat(site.getLabels().isEmpty(), Is.is(false));
 
        for( String label : site.getLabels() ) {
            System.out.printf( "label=[%s]\n", label);
        }
    }

}

File: maven-confluence-reporting-plugin/src/test/kotlin/org/bsc/reporting/SiteTest.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.bsc.reporting;

import org.bsc.confluence.model.Site
import org.junit.Assert
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
import java.nio.file.Paths
import javax.xml.bind.JAXBContext

/**
 *
 * @author bsorrentino
 */
class SiteTest {
    
    
    /**
     * Must be ignored because i got
     * 
     * java.lang.NoClassDefFoundError: org/bsc/maven/reporting/model/Site
     * at org.bsc.reporting.SiteTest.jaxbTest(SiteTest.java:30)
     * 
     * @throws Exception 
     */
    @Test
    @Disabled
    fun jaxbTest() {

        val basepath = Paths.get("src/site/confluence/home.confluence")

        val relativeURI = java.net.URI("template.confluence")

        print("uri=${basepath.toUri().resolve(relativeURI)}")
        
        val jc = JAXBContext.newInstance(Site::class.java)

        val unmarshaller = jc.createUnmarshaller()

        val result = unmarshaller.unmarshal( this::class.java.classLoader.getResourceAsStream("site.xml"));

        assertNotNull(result)
        assertTrue( result is Site)

        val site = result as Site
        
        //site.setBasedir( basepath )

        assertNotNull(site.home.uri)
        assertEquals("home", site.home.name)
        Assert.assertNotNull(site.home)
        assertNotNull(site.home.attachments)
        assertFalse(site.home.attachments.isEmpty())
        assertNotNull(site.home.children)
        assertFalse(site.home.children.isEmpty())
        assertFalse(site.labels.isEmpty());

        site.labels.forEach {
            print( "label=$it\n")
        }
    }

}
